allennlp/tests/common/from_params_test.py

```diff
@@ -680,3 +680,68 @@ def __init__(self, lazy_object: Lazy[ConstructedObject]):
                 assert second_time.extra == extra_string
 
         Testing.from_params(Params({"lazy_object": {"string": test_string}}))
+
+    def test_iterable(self):
+        from allennlp.common.registrable import Registrable
+
+        class A(Registrable):
+            pass
+
+        @A.register("b")
+        class B(A):
+            def __init__(self, size: int) -> None:
+                self.size = size
+
+        class C(Registrable):
+            pass
+
+        @C.register("d")
+        class D(C):  # noqa
```

Comment:
Why do you need all of these `# noqa` statements?

Reply:
I can remove them. It was just that I have warnings of unused class or wrong types.

Reply:
Due to local IDE stuff that isn't reflected in our CI?

Reply:
Yes.

Reply:
I think our policy should be that directives like this that get checked in should strictly be for things that our CI does.  Otherwise it could eventually become a jumbled mess with people adding things for their favorite checker that we're not actually using.

Reply:
Makes sense :+1: